internal/plugin/grpc_provider.go

```diff
@@ -79,12 +79,14 @@ func (p *GRPCProvider) GetProviderSchema() (resp providers.GetProviderSchemaResp
 	defer p.mu.Unlock()
 
 	// check the global cache if we can
-	if !p.Addr.IsZero() {
+	if !p.Addr.IsZero() && resp.ServerCapabilities.GetProviderSchemaOptional {
```

Comment:
this breaks the caching. `resp` is the zero value, i.e. `GetProviderSchemaOptional` is always false.

Reply:
That is intentional. Caching via `GetProviderSchemaOptional` must be opt-in to maintain current behavior with existing provider releases.